debezium-testing/debezium-testing-testcontainers/src/test/java/io/debezium/testing/testcontainers/DebeziumContainerTest.java

```diff
@@ -71,11 +73,21 @@ public static void startContainers() {
     public void canRegisterConnector() throws Exception {
         debeziumContainer.registerConnector("my-connector-1", getConfiguration(1));
 
-        String status = executeHttpRequest(debeziumContainer.getConnectorStatus("my-connector-1"));
-
-        assertThat(JsonPath.<String> read(status, "$.name")).isEqualTo("my-connector-1");
-        assertThat(JsonPath.<String> read(status, "$.connector.state")).isEqualTo("RUNNING");
-        assertThat(JsonPath.<String> read(status, "$.tasks[0].state")).isEqualTo("RUNNING");
+        // task initialization happens asynchronously, so we might have to retry until the task is RUNNING
+        Awaitility.await()
+                .pollInterval(Duration.ofMillis(250))
+                .atMost(Duration.ofSeconds(30))
+                .ignoreException(PathNotFoundException.class)
+                .until(
```

Comment:
If there is a chance the path will exist but contain incorrect/outdate value for a a period of time, this will fail with first failed assertion. The correct method here is ```untilAsserted``` which is specifically design to wait for correct assertions...

Reply:
Adjusted accordingly.